tests/test_grids.py

```diff
@@ -385,3 +385,84 @@ def test_advect_nemo(mode):
     pset = ParticleSet.from_list(field_set, ptype[mode], lon=[lonp], lat=[latp])
     pset.execute(AdvectionRK4, runtime=delta(days=2), dt=delta(hours=6))
     assert abs(pset[0].lat - latp) < 1e-3
+
+
+@pytest.mark.parametrize('mode', ['scipy', 'jit'])
+def test_cgrid_uniform_2dvel(mode):
+    lon = np.array([[0, 2], [.4, 1.5]])
+    lat = np.array([[0, -.5], [.8, .5]])
+    U = np.array([[-99, -99], [4.4721359549995793e-01, 1.3416407864998738e+00]])
+    V = np.array([[-99, 1.2126781251816650e+00], [-99, 1.2278812270298409e+00]])
+
+    dimensions = {'lat': lat, 'lon': lon}
+    data = {'U': np.array(U, dtype=np.float32), 'V': np.array(V, dtype=np.float32)}
+    fieldset = FieldSet.from_data(data, dimensions, mesh='flat')
```

Comment:
also test this on spherical mesh?

Reply:
I can design such test. The problem is that I don't have a velocity dataset. I did create this one.
I'll test by using the velocities from the 2d nemo zonal velocity (so it will be z-grid which I'll artificially move to s-grid).

Reply:
Do you still want to add this test of `mesh=spherical`? I think we should, also because spherical meshes are more common in our applications than flat meshes. 
Once that gets tested, I'm happy for this to be merged into `master`